### PR TITLE
Location listing

### PR DESCRIPTION
**Pull request by LidorLapid**

### CODE REVIEW

File: _includes/elements/location-listing.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        <div>
+          <button type="submit" class="button button--teal location-listing-button flex-item--no-shrink flex-item--grow postfix">Search</button>
+        </div>
+      </div>
+    </div>
+
+    <div class="filters column" data-filters-body-container>
+        <div class="filters__button">
+          <button class="button button--outline button--outline-teal button--filter-icon location-listing-button" data-open-filters>Filter</button>
```
Comment (natmunckton) [inline _includes/elements/location-listing.html]: Is location-listing-button too specific a name?

File: _pages/proving-grounds/contact-a-service.html [hunk 2]
```diff
@@ -24,19 +24,19 @@ <h2 class="impact-strip__heading">General enquiries</h2>
...
+          <dd class="definition-list__description definition-list--double-line-spacing">
             <a class="definition-list__anchor" href="mailto:County.hall@kent.gov.uk">county.hall@kent.gov.uk</a>
           </dd>
           <dt class="definition-list__title">Call:</dt>
-          <dd class="definition-list__description">03000 41 41 41</dd>
+          <dd class="definition-list__description definition-list--double-line-spacing">03000 41 41 41</dd>
           <dt class="definition-list__title">Text relay:</dt>
-          <dd class="definition-list__description">18001 03000 41 41 41</dd>
+          <dd class="definition-list__description definition-list--double-line-spacing">18001 03000 41 41 41</dd>
           <dt class="definition-list__title">Address:</dt>
```
Comment (natmunckton) [inline _pages/proving-grounds/contact-a-service.html]: The definition-list--bottom-border class is missing from the dd's. I've reinstated them in master.

File: _pages/proving-grounds/contact-a-service.html [hunk 3]
```diff
@@ -24,19 +24,19 @@ <h2 class="impact-strip__heading">General enquiries</h2>
 
     <div class="row impact-strip__content">
       <div class="medium-12 column">
-        <dl class="definition-list definition-list--large definition-list--two-columns definition-list--light-text">
+        <dl class="definition-list definition-list--large definition-list--two-columns definition-list--white-text">
           <dt class="definition-list__title">Email:</dt>
-          <dd class="definition-list__description">
+          <dd class="definition-list__description definition-list--double-line-spacing">
             <a class="definition-list__anchor" href="mailto:County.hall@kent.gov.uk">county.hall@kent.gov.uk</a>
```
Comment (natmunckton) [inline _pages/proving-grounds/contact-a-service.html]: The definition-list--bottom-border class is missing from the dd's. I've reinstated them in master.

File: assets/scss/partials/elements/_definition-list.scss [hunk 4]
```diff
@@ -52,16 +51,16 @@
...
     @include breakpoint(medium) {
-      content: '';
-      display: inline-block;
-      height: 1px;
-      padding-bottom: 40px;
+      content: '' !important;
+      display: inline-block !important;
+      height: 1px !important;
+      padding-bottom: 40px !important;
     }
```
Comment (natmunckton) [inline assets/scss/partials/elements/_definition-list.scss]: Are these !importants needed?

File: .gitignore [hunk 5]
```diff
@@ -16,6 +16,7 @@ assets/js/polyfills.js
 assets/js/main-legacy.js
 assets/js/vendor.js
 assets/js/vendor-legacy.js
+assets/js/location-listing.js
```
Comment (wellhairy) [inline .gitignore]: Is their a plan to move this to a dist folder before merging r should we consumed this piece of work in another branch?

File: _config.build.yml [hunk 6]
```diff
@@ -3,6 +3,7 @@ destination: ./_site
 
 defaults:
   paths:
+    assets: /assets
```
Comment (wellhairy) [inline _config.build.yml]: I don't think this is being used anymore

File: _pages/proving-grounds/contact-a-service.html [hunk 7]
```diff
@@ -24,19 +24,19 @@ <h2 class="impact-strip__heading">General enquiries</h2>
...
+          <dd class="definition-list__description definition-list--double-line-spacing">03000 41 41 41</dd>
           <dt class="definition-list__title">Text relay:</dt>
-          <dd class="definition-list__description">18001 03000 41 41 41</dd>
+          <dd class="definition-list__description definition-list--double-line-spacing">18001 03000 41 41 41</dd>
           <dt class="definition-list__title">Address:</dt>
-          <dd class="definition-list__description">County Hall, Maidstone, ME14 1XQ</dd>
+          <dd class="definition-list__description definition-list--double-line-spacing">County Hall, Maidstone, ME14 1XQ</dd>
           <dt class="definition-list__title">Office hours:</dt>
-          <dd class="definition-list__description">Monday to Friday, 8am to 6pm</dd>
+          <dd class="definition-list__description definition-list--double-line-spacing">Monday to Friday, 8am to 6pm</dd>
```
Comment (wellhairy) [inline _pages/proving-grounds/contact-a-service.html]: This double line spacing class might need renaming as It  effects the element above it. Not to mention that I should have really abstracted this class off from master.

File: assets/images/icons/filter.svg [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="21" height="20" viewBox="0 0 21 20">
+  <title>filter</title>
+  <path d="M4.86,6H17.33l-5.23,5.43L12,18.9A1.12,1.12,0,0,1,10.9,20h0a1.12,1.12,0,0,1-1.09-.88L8.21,9.83Z" fill="#0d3a44" fill-rule="evenodd"/>
+  <path d="M1,2.07V1H20l-.22,1.07-6.35,7.8-.53,4.27L8.67,17,7.88,9.87Z" fill="none" stroke="#1f8298" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" fill-rule="evenodd"/>
+</svg>
```
Comment (wellhairy) [inline assets/images/icons/filter.svg]: This isn't being used. I'll look into applying it before we go live

File: assets/js/entry.js [hunk 9]
```diff
@@ -3,6 +3,7 @@ import kcc from './modules/legacy';
 import * as globals from './globals';
 import LazyLoader from './modules/lazyloader';
 
+
```
Comment (wellhairy) [inline assets/js/entry.js]: Can we loose this line

File: _pages/proving-grounds/contact-a-service.html [hunk 10]
```diff
@@ -24,19 +24,19 @@ <h2 class="impact-strip__heading">General enquiries</h2>
 
     <div class="row impact-strip__content">
       <div class="medium-12 column">
-        <dl class="definition-list definition-list--large definition-list--two-columns definition-list--light-text">
```
Comment (LidorLapid) [inline _pages/proving-grounds/contact-a-service.html]: We use a --light-text modifier in cards. The text colour there is also white, so can we make this consistent one way or the other?

File: assets/js/modules/modal.js [hunk 11]
```diff
@@ -124,25 +117,28 @@ export default class Modal {
...
   close() {
+    this.removeEventListeners();
     this.elements.pageOverlay.classList.add(`${this.options.className}--out`);
     this.elements.pageOverlay.classList.remove(`${this.options.className}--active`);
+
     setTimeout(() => {
+      if (this.options.onclose) {
+        this.options.onclose();
+      }
+
```
Comment (wellhairy) [inline assets/js/modules/modal.js]: what does this do exactly? Is this just part of the on close callback?

File: assets/js/modules/utilities.js [hunk 12]
```diff
@@ -74,3 +74,8 @@ export function isDevEnvironment() {
   return (isKentGov() && window.location.href.indexOf('env=dev') > -1) ||
     (document.domain === 'localhost');
 }
+
+export function parseStringAsXml(xmlString) {
+  let parser = new DOMParser();
+  return parser.parseFromString(xmlString, 'application/xml');
+}
```
Comment (wellhairy) [inline assets/js/modules/utilities.js]: This should have been done of master but ignoring that does it need any kind of extra testing?

File: assets/scss/partials/_variables.scss [hunk 13]
```diff
@@ -49,6 +49,7 @@ $text-size-small: 16px;
 $text-size-base: 18px;
 $text-size-medium: 21px;
 $text-size-large: 24px;
+$text-size-xlarge: 28px;
```
Comment (wellhairy) [inline assets/scss/partials/_variables.scss]: Ah ha! I new we had one of these somewhere. Again in our hast we need to make sure we do this lkind of thing in master and pull in.

File: assets/scss/partials/elements/_definition-list.scss [hunk 14]
```diff
@@ -52,16 +51,16 @@
...
+  &--double-line-spacing:before {
     @include breakpoint(medium) {
-      content: '';
-      display: inline-block;
-      height: 1px;
-      padding-bottom: 40px;
+      content: '' !important;
+      display: inline-block !important;
+      height: 1px !important;
+      padding-bottom: 40px !important;
```
Comment (wellhairy) [inline assets/scss/partials/elements/_definition-list.scss]: This level of abstraction (even though i did this) needs to be made in master

File: assets/scss/partials/elements/_maps.scss [hunk 15]
```diff
@@ -1,55 +0,0 @@
...
-	@include breakpoint(medium) {
-		.show-button {
-			display: none;
-		}
-
-		.nested-map {
-			display: block;
-		}
-	}
-}
```
Comment (wellhairy) [inline assets/scss/partials/elements/_maps.scss]: Do we need to test if this was ever needed?

File: assets/scss/partials/structure/_cards.scss [hunk 16]
```diff
@@ -66,6 +66,22 @@
...
+
+    &:last-child {
+      margin-bottom: $margin-bottom;
+    }
+  }
+
+  &--listing-featured {
+    background-color: $overcast-teal;
+  }
+
```
Comment (wellhairy) [inline assets/scss/partials/structure/_cards.scss]: Did we agree that we where keeping this class name for the listings and should this be specific to location lsting?

File: assets/scss/partials/structure/_layouts.scss [hunk 17]
```diff
@@ -1,27 +1,49 @@
...
   }
 
   &--no-shrink{
     flex-shrink: 0;
   }
+
+  &--shrink{
+    flex-shrink: 1;
+  }
 }
```
Comment (wellhairy) [inline assets/scss/partials/structure/_layouts.scss]: Do we need to think about these class names any more, I think we agreed earlier that these shouldn't be modifiers but standalone classes?

File: assets/scss/partials/structure/_layouts.scss [hunk 18]
```diff
@@ -49,3 +71,39 @@
...
+    border-color: darken($white, 20);
+  }
+
+  &--remove-at-small {
+    @include breakpoint(small only){
+      border-width: 0px;
+    }
+  }
+}
+
```
Comment (wellhairy) [inline assets/scss/partials/structure/_layouts.scss]: Thoughts about thee, again should they be done differently?

File: _pages/proving-grounds/contact-a-service.html (hunk 10 above)
Comment (wellhairy) [inline _pages/proving-grounds/contact-a-service.html]: Sure, do they do the same thing? if they do I'd say go with  the method introduced by cards

File: _includes/elements/location-listing.html (hunk 1 above)
Comment (wellhairy) [inline _includes/elements/location-listing.html]: This is a stop gap class before we merge in the buttons refactor

Comment (wellhairy) [inline _includes/elements/location-listing.html]: I also need to add a class for the icon styling

**Pull request merged**